lshw: B.02.18 -> 02.19

thiagokokada opened the pull request:
###### Description of changes

- Bump version
- Switch to `fetchFromGitHub` to get updates from @r-ryantm bot
- Remove patches (they're merged upstream)
- Update dependencies
- Fix version workaround (not necessary anymore)
- Add myself as maintainer
- Fix the version for repology.org (see commit message for details)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

thiagokokada committed:
lshw: B.02.18 -> B.02.19

- Bump version
- Switch to `fetchFromGitHub` to get updates from @r-ryantm bot
- Remove patches (they're merged upstream)
- Update dependencies
- Fix version workaround (not necessary anymore)
- Add myself as maintainer

superherointj reviewed (approved):
`lshw` (CLI) runs for me. GUI not tested.

thiagokokada committed:
lshw: rename version to fix repology listing

`lshw` is not showing in the nixpkgs-unstable listings in repology, but
it is showing in the previous stable release (release-21.11). The reason
for this was a change in 8d0267dc8f02e010785d6f90573ea1135b7957e7, that
fixed the `lshw` version to include its letter prefix.

However, the way the version is computed for repology is to parse the
`name` attr instead, separating it in `pname` and `version`. The
function that does this (`builtins.parseDrvName`) considers anything
that is a `name` everything up to the first dash followed by a digit.
Because the `version` includes the letter `B` as prefix, it them ends up
splitting it wrong.

See https://github.com/NixOS/nix/pull/4463 for a proper fix, but for now
let's just "fix" this by not including the prefix in the `version`.

SuperSandro2000 commented:
@ofborg eval

thiagokokada reviewed (commented):
Sorry, just realized that the `lshw -version` would report the wrong version. Now this is fixed.

SuperSandro2000 commented on pkgs/tools/system/lshw/default.nix:
```diff
@@ -1,40 +1,40 @@
-  makeFlags = [ "PREFIX=$(out)" ];
+  makeFlags = [
+    "PREFIX=$(out)"
+    "VERSION=${actualVersion}"
```
```suggestion
    "VERSION=${src.rev}"
```

SuperSandro2000 commented on pkgs/tools/system/lshw/default.nix:
```diff
@@ -1,40 +1,40 @@
+  # Fix repology.org by not including the prefixed B, otherwise the `pname` attr
+  # gets filled as `lshw-B.XX.XX` in `nix-env --query --available --attr nixpkgs.lshw --meta`
+  # See https://github.com/NixOS/nix/pull/4463 for a definitive fix
+  version = "02.19";
+  actualVersion = "B.02.19";
```
```suggestion
  version = "02.19";
  actualVersion = "B.${version}";
```
we don't want to duplicate this but we also don't need this here.

thiagokokada replied:
Done.
